src/redux-middleware/pullQueue.ts

```diff
@@ -178,6 +180,11 @@ const pullQueueMiddleware: ThunkMiddleware<State> = ({ getState, dispatch }) =>
     syncStatusStore.update({ isPulling: false })
     pulling.delete(extendedPullQueueFiltered)
 
+    // scroll cursor into view if the pull was the result of navigation to a thought
+    if (navigatedStore.getState()) {
+      scrollCursorIntoView()
+    }
+
```

Comment:
It would be better to move this into the [scrollCursorIntoView middleware](https://github.com/cybersemics/em/blob/bbb87809aba6cbb002cdc3cdfa4c300f45cf2b9c/src/redux-middleware/scrollCursorIntoView.ts), so the logic is more centralized, and the pullQueue doesn't need to know about the navigatedStore or cursor.

We can't directly detect when a `pull` completes from the other middleware, but we have a couple other options:

1. Since a `pull` always dispatches [updateThoughts](https://github.com/cybersemics/em/blob/bbb87809aba6cbb002cdc3cdfa4c300f45cf2b9c/src/actions/updateThoughts.ts), you can call `scrollCursorIntoView` in response by checking if `action.type === 'updateThoughts'` . This was suggested in the original issue. Did you run into problems with this approach?
2. Since a `pull` is always followed by `syncStatusStore.update({ isPulling: false })`, you can subscribe to the `syncStatusStore` and call `scrollCursorIntoView` when `isPulling` changes to false.